### PR TITLE
New-style tasks/workflows use user exception scope

### PR DESCRIPTION
**Pull request by slai**
# TL;DR

New-style tasks/workflows are currently not wrapped in a user exception scope, so all exceptions in user code are treated as unknown system errors which are automatically retried. Further context - https://flyte-org.slack.com/archives/CREL4QVAQ/p1625581856326400

This PR restores the user exception scope to new-style tasks/workflows so exceptions in user code are not retried unless FlyteRecoverableException is raised.

## Type
 - [x] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [ ] Code completed
 - [ ] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
 _How did you fix the bug, make the feature etc. Link to any design docs etc_

### CODE REVIEW

File: flytekit/bin/entrypoint.py [hunk 1]
```diff
@@ -96,7 +96,11 @@ def _dispatch_execute(
             c: OR if an unhandled exception is retrieved - record it as an errors.pb
     """
     output_file_dict = {}
-    try:
+
+    @_scopes.system_entry_point
```
Comment (slai) [inline flytekit/bin/entrypoint.py:100]: Although `_dispatch_execute` is already being executed in a system exception scope, that scope is too far up the stack to wrap exceptions raised in this function so the subsequent except blocks below can handle them.
